tests/functional/aws-node-sdk/lib/utility/bucket-util.js

```diff
@@ -1,139 +1,146 @@
-const AWS = require('aws-sdk');
-AWS.config.logger = console;
-const { S3 } = require('aws-sdk');
+const {
+    S3Client,
+    HeadBucketCommand,
+    CreateBucketCommand,
+    DeleteBucketCommand,
+    ListObjectVersionsCommand,
+    DeleteObjectCommand,
+    ListBucketsCommand,
+} = require('@aws-sdk/client-s3');
 const projectFixture = require('../fixtures/project');
 const getConfig = require('../../test/support/config');
 
 class BucketUtility {
-    constructor(profile = 'default', config = {}) {
+    constructor(profile = 'default', config = {}, unauthenticated = false) {
         const s3Config = getConfig(profile, config);
-
-        this.s3 = new S3(s3Config);
-        this.s3.config.setPromisesDependency(Promise);
-        this.s3.config.update({
-            maxRetries: 0,
-        });
+        if (unauthenticated) {
+            this.s3 = new S3Client({
+                ...s3Config,
+                credentials: { accessKeyId: '', secretAccessKey: '' },
+                forcePathStyle: true,
+                signer: { sign: async request => request },
+             });
+        }
+       else {
+            this.s3 = new S3Client({
+                ...s3Config,
+                maxAttempts: 0,
+             });
+       }
```

Comment:
```suggestion
        if (unauthenticated) {
            s3Config = {
              ...s3Config,
              credentials: { accessKeyId: '', secretAccessKey: '' },
              forcePathStyle: true,
              signer: { sign: async request => request },
            };
        }

            this.s3 = new S3Client({
                ...s3Config,
                maxAttempts: 0,
             });
```

Pretty sure that `maxAttempts: 0,` is working also for unauthenticated client ?